### PR TITLE
feat(angular): add standalone modal

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: N/A

---------

## What is the current behavior?

Modal cannot be used as a standalone component

## What is the new behavior?

- Added modal as a standalone component
- Added a test
- Lazy and standalone modal now extend from the same base implementation

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/angular/common/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// TODO: Is there a way we can grab this from angular-component-lib instead?
```
Comment (liamdebeasi) [inline packages/angular/common/src/utils.ts:1]: See https://github.com/ionic-team/ionic-framework/pull/27883#discussion_r1277684368

**Review by thetaPC — APPROVED** [review]
LGTM

File: packages/angular/common/src/overlays/modal.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  /**
+   * Emitted after the modal has dismissed.
+   */
+  ionModalDidDismiss: EventEmitter<CustomEvent>;
+  /**
+   * Emitted after the modal breakpoint has changed.
+   */
+  ionBreakpointDidChange: EventEmitter<CustomEvent<ModalBreakpointChangeEventDetail>>;
+  /**
+   * Emitted after the modal has presented. Shorthand for ionModalWillDismiss.
```
Comment (sean-perkins) [inline packages/angular/common/src/overlays/modal.ts:36]:
```suggestion
   * Emitted after the modal has presented. Shorthand for ionModalDidPresent.
```

**Review by sean-perkins — CHANGES_REQUESTED** [review]
Similar feedback to popover. Implementation changes look good. Just one minor documentation tweak.

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**